api/src/utils/verify-session-jwt.ts

```diff
@@ -15,7 +15,7 @@ export async function verifySessionJWT(payload: DirectusTokenPayload) {
 		.from('directus_sessions')
 		.where({
 			token: payload['session'],
-			user: payload['id'],
+			user: payload['id'] || null,
```

Comment:
We should include the share ID in the where clause

```suggestion
			user: payload['id'] || null,
			share: payload['share'] || null,
```

Reply:
Note that #22693 is required for that

Reply:
Thank you for the review! It's fixed now :)